pysat/tests/test_orbits.py

```diff
@@ -12,7 +12,25 @@
 import pysat
 
 
+def filter_data(inst, times=None):
+    """Remove data from instrument, simulating gaps in the dataset"""
+
+    if times is None:
+        times = [[dt.datetime(2009, 1, 1, 1), dt.datetime(2009, 1, 1, 2)],
+                 [dt.datetime(2009, 1, 1, 10), dt.datetime(2009, 1, 1, 12)],
+                 [dt.datetime(2009, 1, 1, 22), dt.datetime(2009, 1, 2, 2)],
+                 [dt.datetime(2009, 1, 4), dt.datetime(2009, 1, 6)]
+                 ]
```

Comment:
Some specific edge cases are getting lost.
- single orbit gap
- orbit < gap < 2 orbits
- gap across daybreak, gap > 2 orbits, , no extra data for last orbit on 1st, no data in first orbit time on 2nd
- gap > 1 day, data in first orbit time on 15th, no data on 14th
- gap > 1 day, only a little bit of last orbit on 20th and a little bit of first orbit on 25th available.
- orbit < gap < 2 orbits, daybreak check, 
```
    times = [[dt.datetime(2009, 1, 1, 1, 37), dt.datetime(2009, 1, 1, 3, 14)],
              [dt.datetime(2009, 1, 1, 10), dt.datetime(2009, 1, 1, 12)],
              [dt.datetime(2009, 1, 1, 22), dt.datetime(2009, 1, 2, 2)],
              [dt.datetime(2009, 1, 13), dt.datetime(2009, 1, 15)],
              [dt.datetime(2009, 1, 20, 1), dt.datetime(2009, 1, 25, 23)],
              [dt.datetime(2009, 1, 25, 23, 30), dt.datetime(2009, 1, 26, 3)]
```

Reply:
Do we need a month of data here, or can we compress into a week and capture all these use cases?